### PR TITLE
Extend client interface (sync) if extension methods exist in classpath

### PR DESCRIPTION
**Pull request by cenedhryn**

### CODE REVIEW

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetUtils.java [hunk 1]
```diff
@@ -109,4 +112,15 @@ public static JavaFile buildJavaFile(ClassSpec spec) {
         return builder.build();
     }
 
+    public static Optional<Class<?>> findExtensionClass(ClassName clientInterfaceName, Class<?> clazz) {
```
Comment (Bennett-Lynch) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetUtils.java:115]: Can `clientInterfaceName` be inferred from `clazz`?

File: codegen/src/test/java/software/amazon/awssdk/codegen/poet/utils/SomeClientExtensionMethods.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.poet.utils;
+
+/**
+ * Test class used by {@link PoetUtilsTest}
+ */
+public class SomeClientExtensionMethods {
```
Comment (Bennett-Lynch) [inline codegen/src/test/java/software/amazon/awssdk/codegen/poet/utils/SomeClientExtensionMethods.java:21]: Should we require that this be an interface?

**Review by Bennett-Lynch — APPROVED** [review]

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetUtils.java (hunk 1 above)
Comment (cenedhryn) [reply codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetUtils.java:115]: No, the clazz is determining how the class loader works. I meant to rename it to make it clearer lm do that.

File: codegen/src/test/java/software/amazon/awssdk/codegen/poet/utils/SomeClientExtensionMethods.java (hunk 2 above)
Comment (cenedhryn) [reply codegen/src/test/java/software/amazon/awssdk/codegen/poet/utils/SomeClientExtensionMethods.java:21]: Hah, good point. I had mixed and matched in my tests but IRL it doesn't work if it's a class...

**Review by Bennett-Lynch — APPROVED** [review]

**Pull request merged**